### ISSUE TITLE
Page reloads first time using `push()`

### ISSUE THREAD

**Issue by avimar** [ItalyPaleAle/svelte-spa-router#121]
I don't think I'm doing anything weird, but when clicking a link, the hashbang in the URL changes but then the entire page reloads (sometimes on the same page.)

This seems to have to do with a ? in the url.

1) Live example: https://freedomhot.com/sim-activate.html
1b) auto redirects to https://freedomhot.com/sim-activate.html#1
```
import {push, replace, location } from 'svelte-spa-router';
if($location=='/') replace('/1');//default to page one.
```
2) I click a box, then click next. This calls `push('/2');`
3) What I expect: go to the next page, the URL changes to https://freedomhot.com/sim-activate.html#2
But what happens, as per dev tools console:
`Navigated to https://freedomhot.com/sim-activate.html?` and in the networktab, I see everything reloaded.
 The url now reads https://freedomhot.com/sim-activate.html?#2 with the `?`.

Now going back and forth is fine.

Oddly, when I try again with `https://freedomhot.com/sim-activate.html#2` and use my regular link `<a href="#/1" class="btn btn-success" style="float:left">` it's fine. It's only with the included `push('/3');` that cause this issue.

I'm on the latest chrome. Is there something I'm not understanding about this?
I'd wonder if google analytics or hotjar is interfering with it, but looks like ublock is blocking them.

**Comment by ItalyPaleAle** [comment]
It's hard for me to help without having access to the source code. I've tried taking a look at it and one weird thing I could notice is the variable named `$form_activate`: normally, the $ has a special meaning in Svelte, so it *might* cause some issues.

However, I can't debug without access to the source code.

I do not believe this is an issue with the router itself however. The fact that it works after the page refreshes makes me think that more…

**Comment by avimar** [comment]
Wow OK, so I figured it out.

1) A basic reproduction worked just fine, with the `$location` and `push`.
2) I tried cutting code from my site, but it still didn't work. (GIT is awesome. Just delete stuff and experiment and don't worry, you can simply revert it later!)
After deleting nearly everything, the `submit` button caught my eye... and then I figure it out...

**The issue**: my buttons were doing a submit which caused chrome to reload the page.. but only 1x until a `?` was in the url.
Solution:
a) On my `<button>`, use `type="button"` and not the assumed `type="submit"` inside a `<form>`
b) Or: on `<button>` that I want to actually use as a submit-on-enter, use `on:click|preventDefault={handler}`.
I went out of my way to include a `<form>` element for the keyboard and accessibility functionality, which made the browser assume the buttons were `submit`.

Seems this also fixes my #120 ie11 issue with "[after polyfill] then we don't get any errors, we just get a redirect to the base page like other time." (I didn't even try adding the polyfill now... babel should have handled that.)


Seems like a pretty simple issue, if you know about buttons... (I checked, since I didn't set `action='post'` it was still a `GET` so even if `Method` would have been shown in dev-tools, it wouldn't have been obvious.)

Thanks for your time! And I hope this helps someone else in the future....
